controllers/change_coordinators.go

```diff
@@ -85,20 +85,37 @@ func (c ChangeCoordinators) Reconcile(r *FoundationDBClusterReconciler, context
 		log.Info("Changing coordinators", "namespace", cluster.Namespace, "cluster", cluster.Name)
 		r.Recorder.Event(cluster, "Normal", "ChangingCoordinators", "Choosing new coordinators")
 
+		coordinatorCount := cluster.DesiredCoordinatorCount()
 		candidates := make([]localityInfo, 0, len(status.Cluster.Processes))
-		for _, process := range status.Cluster.Processes {
-			eligible := !process.Excluded && isStateful(process.ProcessClass) && !cluster.InstanceIsBeingRemoved(process.Locality[FDBInstanceIDLabel])
-			if eligible {
-				candidates = append(candidates, localityInfoForProcess(process))
+		selectCandidates := func(candidates []localityInfo, class fdbtypes.ProcessClass) []localityInfo {
```

Comment:
In general this looks good to me. Could you move the coordinator selection into its own method and add some unit tests?

Reply:
I'll look for a timeslice to do this. I've added the PDB example and pushed.

Reply:
@johscheuer this is now ready for review. sorry for the delay - timeslices are precious :)

Reply:
Actually sorry, I see the examples are tested now, so I'll fix that up too.

Reply:
@johscheuer actually actually ready now :)